src/Native/Runtime/portable.cpp

```diff
@@ -208,16 +218,39 @@ COOP_PINVOKE_HELPER(Object *, RhpNewFastAlign8, (EEType* pEEType))
 
     int requiresPadding = ((uint32_t)result) & 7;
     size_t paddedSize = size;
-    if (requiresPadding) paddedSize += 12;
+    if (requiresPadding) 
+    {
+#ifdef HOST_64BIT
```

Comment:
RhpNewFastAlign8 does not make sense for 64-bit. And if it did, the constant 12 would make sense for it. I think you can ifdef out the whole method for 64-bit.